src/core/project/qgsproject.h

```diff
@@ -885,8 +886,27 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
      * Make sure that this is only called when all layers are not in edit mode.
      *
      * \since QGIS 2.16
+     * \deprecated QGIS 3.24 use setTransactionMode instead
      */
-    void setAutoTransaction( bool autoTransaction );
+    Q_DECL_DEPRECATED void setAutoTransaction( bool autoTransaction ) SIP_DEPRECATED;
+
+    /**
+     * Returns the transaction mode
+     *
+     * \see Qgis::TransactionMode
+     * \since QGIS 3.24
+     */
+    Qgis::TransactionMode transactionMode() const;
+
+    /**
+     * Set transaction mode
+     *
+     * Make sure that this is only called when all layers are not in edit mode.
```

Comment:
Should an exception be raised if this condition is not met?

Reply:
Probably yes, but looking at the code I don't understand exactly why. Probably unsaved modification would be lost when the transaction groups are cleared.

Reply:
I have made the function returning false if at leas one layer is in edit mode.